repose-aggregator/core/valve/src/test/scala/org/openrepose/spring/ValveTestModeRunnerTest.scala

```diff
@@ -0,0 +1,415 @@
+  }
+
+  def getValvePortMXBean: ValvePortMXBean = {
+    val mbs = ManagementFactory.getPlatformMBeanServer
+    val name = new ObjectName(ValvePortMXBean.OBJECT_NAME)
+    if (mbs.isRegistered(name)) {
+      println("Getting the registered mbean!")
+      val thing = JMX.newMBeanProxy(mbs, name, classOf[ValvePortMXBean])
+      thing
+    } else {
+      fail("Unable to get ValvePort MX Bean!")
+    }
+  }
+
+  def updateSystemModel(resource: String): UpdateListener[SystemModel] = {
+    val systemModelListener = fakeConfigService.getListener[SystemModel]("system-model.cfg.xml")
+    val systemModel = Marshaller.systemModel(resource)
+    systemModelListener.configurationUpdated(systemModel)
+    systemModelListener
+  }
+
+  def updateContainerConfig(resource: String): UpdateListener[ContainerConfiguration] = {
+    val containerListener = fakeConfigService.getListener[ContainerConfiguration]("container.cfg.xml")
+    val containerConfig = Marshaller.containerConfig(resource)
+    containerListener.configurationUpdated(containerConfig)
+    containerListener
+  }
+
+  it("has a blocking run method") {
+    val runner = new ValveRunner(fakeConfigService)
+
+    val future = Future {
+      runner.run("/root", false)
+    }
+    future.isCompleted shouldBe false
+
+    runner.destroy()
+    Await.ready(future, 1 second)
+    future.isCompleted shouldBe true
+  }
+
+  it("passes through configRoot and insecure to each node") {
+    pending
+  }
+
+  describe("Starting fresh") {
+    it("Does nothing if the container-config is updated before the system-model") {
+      withRunner() { runner =>
+        runner.getActiveNodes shouldBe empty
+
+        updateContainerConfig("/valveTesting/without-keystore.xml")
+
+        //it should not have triggered any nodes
+        runner.getActiveNodes shouldBe empty
+        getValvePortMXBean.getPort("repose", "repose_node1") shouldBe 0
+      }
+
+    }
+    it("doesn't start nodes if only the system-model has been configured") {
+      withRunner() { runner =>
+        runner.getActiveNodes shouldBe empty
+
+        updateSystemModel("/valveTesting/1node/system-model-1.cfg.xml")
+
+        //it should not have triggered any nodes
+        runner.getActiveNodes shouldBe empty
+        getValvePortMXBean.getPort("repose", "repose_node1") shouldBe 0
+      }
+    }
+    it("Starts up nodes as configured in the system-model when hit with a system model before a container config") {
+      withRunner() { runner =>
+        runner.getActiveNodes shouldBe empty
+
+        updateSystemModel("/valveTesting/1node/system-model-1.cfg.xml")
+        updateContainerConfig("/valveTesting/without-keystore.xml")
+
+        runner.getActiveNodes.size shouldBe 1
+        val port = getValvePortMXBean.getPort("repose", "repose_node1")
+        logger.debug(s"PORT IS: $port")
+        port shouldNot be(0)
+        port shouldNot be(10234) //It shouldn't match exactly what we configured...
```

Comment:
It could be. But it's very much not likely to be. 

On May 8, 2015 3:03:34 PM CDT, Adrian J George III <notifications@github.com> wrote:
>> +        runner.getActiveNodes shouldBe empty
>> +        getValvePortMXBean.getPort("repose", "repose_node1")
>shouldBe 0
>> +      }
>> +    }
>> +    it("Starts up nodes as configured in the system-model when hit
>with a system model before a container config") {
>> +      withRunner() { runner =>
>> +        runner.getActiveNodes shouldBe empty
>> +
>> +       
>updateSystemModel("/valveTesting/1node/system-model-1.cfg.xml")
>> +        updateContainerConfig("/valveTesting/without-keystore.xml")
>> +
>> +        runner.getActiveNodes.size shouldBe 1
>> +        val port = getValvePortMXBean.getPort("repose",
>"repose_node1")
>> +        logger.debug(s"PORT IS: $port")
>> +        port shouldNot be(0)
>> +        port shouldNot be(10234) //It shouldn't match exactly what
>we configured...
>
>why not? Did i miss where you said that the random port couldn't be the
>same?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/rackerlabs/repose/pull/1297/files#r29971812

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.